java/src/com/ibm/streamsx/topology/tuple/SimpleMessage.java

```diff
@@ -0,0 +1,117 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015  
+ */
+package com.ibm.streamsx.topology.tuple;
+
+import java.io.Serializable;
+
+/**
+ * A Simple implementation of {@link Message}
+ */
+public class SimpleMessage implements Serializable, Message {
+    private static final long serialVersionUID = 1L;
+
+    private final String topic;
+    private final String key;
+    private final String message;
+
+    /**
+     * Create a {@code SimpleMessage} that lacks a topic and a key.
+     * 
+     * @param message
+     *            the message
+     */
+    public SimpleMessage(String message) {
+        this(message, null);
+    }
+    /**
+     * Create a {@code SimpleMessage} that lacks a topic.
+     * @param message
+     *            the message
+     * @param key
+     *            Optional key associated with the message.  May be null.
+     */
+    public SimpleMessage(String message, String key) {
+        if (message == null)
+            throw new IllegalArgumentException("message==null");
+        this.topic = null;
+        this.key = key;
+        this.message = message;
+    }
+
+    /**
+     * Create a {@code SimpleMessage} that includes a topic.
+     * @param message
+     *            the message
+     * @param key
+     *            Optional key associated with the message.  May be null.
+     * @param topic
+     *            Kafka topic identifier.
+     * @throws IllegalArgumentException if topic == null
+     * @throws IllegalArgumentException if message == null
+     */
+    public SimpleMessage(String message, String key, String topic) {
+        if (topic == null)
+            throw new IllegalArgumentException("topic==null");
```

Comment:
The idea was that the two param ctor was for the null-topic case and this variant was strictly for the non-null case.  But in practice that enforcement did make use a bit cumbersome... probably w/o any real benefit.  I'll remove the check.  That said, since I'm going back in, I think I'll toss the  and 3 (String) param ctors and add a Builder.